src/utils/graph_building.jl

```diff
@@ -91,20 +92,32 @@ function weights_cutoff(is, js, dists; max_num_nbr = 12, dist_decay_func = inver
 
     # iterate over list of tuples to build edge weights...
     # note that neighbor list double counts so we only have to increment one counter per pair
-    weight_mat = zeros(Float32, num_atoms, num_atoms)
+    weight_mat = zeros(Float64, round(Int,num_atoms), round(Int,num_atoms))
+    weight_mat, longest_dists = _cutoff!(weight_mat,
+                                         dist_decay_func,
+                                         ijd,
+                                         nb_counts,
+                                         longest_dists)
+    # average across diagonal, just in case
+    weight_mat = 0.5 .* (weight_mat .+ weight_mat')
+end
+
+
+function _cutoff!(weight_mat, f, ijd,
+                  nb_counts, longest_dists; max_num_nbr = 12)
+
     for (i, j, d) in ijd
         # if we're under the max OR if it's at the same distance as the previous one
-        if nb_counts[i] < max_num_nbr || isapprox(longest_dists[i], d)
-            weight_mat[i, j] += dist_decay_func(d)
-            longest_dists[i] = d
-            nb_counts[i] += 1
+        if nb_counts[round(Int,i)] < max_num_nbr || isapprox(longest_dists[round(Int,i)], d)
+            weight_mat[round(Int,i), round(Int,j)] += f(d)
+            longest_dists[round(Int,i)] = d
+            nb_counts[round(Int,i)] += 1
         end
     end
```

Comment:
@DhairyaLGandhi does this look cleaner? Or did you have something else in mind?
```julia
    for (i, j, d) in ijd
        (i, j) = round.(Int, [i, j])
        # if we're under the max OR if it's at the same distance as the previous one
        if nb_counts[i] < max_num_nbr || isapprox(longest_dists[i], d)
            weight_mat[i, j] += f(d)
            longest_dists[i] = d
            nb_counts[i] += 1
        end
    end
```

Reply:
I guess so? It's not any better because we don't really need it to be an integer. It's only for CI.

Reply:
I guess non integer indices are odd too, but technically not out of scope (think Interpolations). Best not to conflate it though.